tests/test_circular_imports.py

```diff
@@ -0,0 +1,71 @@
+"""Test to check for circular imports in core components."""
+import asyncio
+import sys
+
+import pytest
+
+
+@pytest.mark.timeout(30)  # cloud can take > 9s
+@pytest.mark.parametrize(
+    "component",
+    (
+        "api",
+        "application_credentials",
+        "auth",
+        "automation",
+        "backup",
+        "bluetooth",
+        "camera",
+        "cloud",
+        "config",
+        "counter",
+        "device_automation",
+        "dhcp",
+        "diagnostics",
+        "file_upload",
+        "frontend",
+        "frontend",
+        "hardware",
+        "hassio",
+        "homeassistant",
+        "http",
+        "input_boolean",
+        "input_button",
+        "input_datetime",
+        "input_number",
+        "input_select",
+        "input_text",
+        "logger",
+        "logger",
+        "lovelace",
+        "mqtt_eventstream",
+        "network",
+        "onboarding",
+        "persistent_notification",
+        "person",
+        "recorder",
+        "repairs",
+        "scene",
+        "schedule",
+        "script",
+        "search",
+        "sensor",
+        "sentry",
+        "ssdp",
+        "system_health",
+        "system_log",
+        "tag",
+        "timer",
+        "usb",
+        "websocket_api",
+        "zeroconf",
+        "zone",
```

Comment:
What keeps this list up to date? Can we populate it automatically?

Reply:
adjusted in 032125fde027cadfa5efdb3f9634743ae630a3ff